### PR TITLE
Include CUD timestamps and initial filtering

### PR DESCRIPTION
**Pull request by Willdotwhite**

### CODE REVIEW

File: app/src/main/kotlin/team/finder/api/teams/Team.kt [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
+    val createdAt: String,
+    var updatedAt: String,
+    var deletedAt: String,
+
+    @Id val id: Long
 ) {
-    constructor(a: String, d: String, s: Int) : this(a, d, s, 0)
+
+    constructor(_author: String, _description: String, _skillsetMask: Int) :
+            this(_author, _description, _skillsetMask, "", "", "", 0)
```
Comment (Boivin92) [inline app/src/main/kotlin/team/finder/api/teams/Team.kt:21]: I'm a newb in Kotlin so can anyone tell me what's the ``this(params)`` do?

Comment (Boivin92) [reply app/src/main/kotlin/team/finder/api/teams/Team.kt:21]: it's part of the constructor, I get that, I just don't understand why it's there

**Review by Boivin92 — APPROVED** [review]

File: app/src/main/kotlin/team/finder/api/teams/Team.kt (hunk 1 above)
Comment (Willdotwhite) [reply app/src/main/kotlin/team/finder/api/teams/Team.kt:21]: This structure is a "secondary constructor" - it allows us to build a constructor that doesn't require all of the values of the base constructor in order to function.

I'm not 100% sure I understand it all, but it looks like it's to do with hydrating the TeamDTO instances?

Comment (Boivin92) [reply app/src/main/kotlin/team/finder/api/teams/Team.kt:21]: Black Magic, got it ;)
Seriously I'll go have a look at the docs, thanks

**Pull request merged**